mediathread/main/views.py

```diff
@@ -787,17 +788,40 @@ def get_success_url(self):
         return reverse('course-invite-complete')
 
 
-class CourseInstructorDashboardView(LoggedInMixin, DetailView):
+class InstructorDashboardView(LoggedInMixin, DetailView):
     model = Course
-    template_name = 'main/course_instructor_dashboard.html'
+    template_name = 'main/instructor_dashboard.html'
 
     def get_context_data(self, *args, **kwargs):
-        ctx = super(CourseInstructorDashboardView, self).get_context_data(
+        ctx = super(InstructorDashboardView, self).get_context_data(
             *args, **kwargs)
         ctx.update({'course': ctx.get('object')})
         return ctx
 
 
+class InstructorDashboardSettingsView(
```

Comment:
The view should check that the logged in user is faculty for the requested course.